api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,461 @@
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// MachineNetwork is the list of IP address pools for machines.
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+
+	// NetworkType is the Container Network Interface (CNI) plug-in to install
+	// The default value is OpenShiftSDN for IPv4, and OVNKubernetes for IPv6 or SNO
+	// +kubebuilder:validation:Enum=OpenShiftSDN;OVNKubernetes
+	// +kubebuilder:default:=OVNKubernetes
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Additional cluster-wide annotations to be applied to the rendered templates
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
+
+	// ClusterLabels is used to assign labels to the cluster to assist with policy binding.
+	// +optional
+	ClusterLabels map[string]string `json:"clusterLabels,omitempty"`
+
+	// InstallConfigOverrides is a generic way of passing install-config parameters.
+	// +optional
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
+
+	// Json formatted string containing the user overrides for the initial ignition config
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// DiskEncryption is the configuration to enable/disable disk encryption for cluster nodes.
+	// +optional
+	DiskEncryption DiskEncryption `json:"diskEncryption,omitempty"`
+
+	// ProxySettings defines the proxy settings used for the install config
+	// +optional
+	ProxySettings aiv1beta1.Proxy `json:"proxy,omitempty"`
+
+	// ExtraManifestsRefs is list of config map references containing additional manifests to be applied to the cluster.
+	// +optional
+	ExtraManifestsRefs []corev1.LocalObjectReference `json:"extraManifestsRef,omitempty"`
+
+	// CPUPartitioning determines if a cluster should be setup for CPU workload partitioning at install time.
+	// When this field is set the cluster will be flagged for CPU Partitioning allowing users to segregate workloads to
+	// specific CPU Sets. This does not make any decisions on workloads it only configures the nodes to allow CPU Partitioning.
+	// The "AllNodes" value will setup all nodes for CPU Partitioning, the default is "None".
+	// +kubebuilder:validation:Enum=None;AllNodes
+	// +kubebuilder:default=None
+	// +optional
+	CPUPartitioning CPUPartitioningMode `json:"cpuPartitioningMode,omitempty"`
+
+	// TODO: is this field still needed?
+	// +optional
+	SiteConfigMap SiteConfigMap `json:"siteConfigMap,omitempty"`
+
+	// NumMasters is the number of master nodes.
+	// +kubebuilder:validation:Enum=1;3
+	// +kubebuilder:default:=1
+	NumMasters uint8 `json:"numMasters,omitempty"`
```

Comment:
Should we use control plane here? I think we're trying to avoid `master` now, right?

Reply:
Agreed! Changed `NumMasters` -> `ControlPlaneAgents` and `NumWorkers` -> `WorkerAgents`.